### PR TITLE
Fix failing PR workflows

### PR DESCRIPTION
**Pull request by jpreese**
PRs are failing on the E2E tests due to testing on v1.25+ versions of Kubernetes due to the policy API no longer existing. (ref: https://github.com/plexsystems/konstraint/pull/329)

There is also an issue with documentation generation where `go run` fails because the `go.mod` is outdated, but it shouldn't be used for this process.

### CODE REVIEW

File: .github/workflows/cron_e2e.yaml [hunk 1]
```diff
@@ -62,8 +64,8 @@ jobs:
           ./konstraint create -o e2e-resources test
 
       - name: create kind cluster
-        run: kind create cluster
-      
+        run: kind create cluster --image=kindest/node:v1.24.0
```
Comment (jalseth) [inline .github/workflows/cron_e2e.yaml:67]: I think another alternative would be to disable GK from trying to create PSP resources, via `--set psp.enabled=false` in the Helm install command. Either will unblock us in the short term though, up to you.

https://github.com/open-policy-agent/gatekeeper/blob/8a3e2159a0e8be138d46c4a199829a5b8d03d5d5/charts/gatekeeper/values.yaml#L235

File: .github/workflows/pull_request.yaml [hunk 2]
```diff
@@ -140,8 +146,8 @@ jobs:
           ./konstraint create -o e2e-resources test
 
       - name: create kind cluster
-        run: kind create cluster
-      
+        run: kind create cluster --image=kindest/node:v1.24.0
```
Comment (jalseth) [inline .github/workflows/pull_request.yaml:149]: Same comment here.

File: .github/workflows/update_cli_docs.yaml [hunk 3]
```diff
@@ -22,9 +23,7 @@ jobs:
         uses: actions/checkout@v3
 
       - name: generate docs
-        run: |
-          go mod download
-          go run .github/gendocs/konstraint.go
+        run: go run -mod=mod .github/gendocs/konstraint.go
```
Comment (jalseth) [inline .github/workflows/update_cli_docs.yaml:26]: Consider splitting these changes to a separate PR.

**Review by jalseth — APPROVED** [review]

File: .github/workflows/cron_e2e.yaml (hunk 1 above)
Comment (jpreese) [reply .github/workflows/cron_e2e.yaml:67]: Oh! I didn't even realize that was configurable. That would probably be better, I'll give it a shot.

Comment (jpreese) [reply .github/workflows/cron_e2e.yaml]: Worked! 🎉

File: .github/workflows/update_cli_docs.yaml (hunk 3 above)
Comment (jpreese) [reply .github/workflows/update_cli_docs.yaml:26]: Made an attempt at this, but then the PR failed without the change -- so to be able to merge with everything passing, we'd need to include it.

I did some rewording of the PR title and the commit messages to paint a better picture.

**Pull request merged**